spec/integration/server_create_integ_spec.rb

```diff
@@ -56,34 +75,61 @@ def validate_output(shell, params)
     }
   end
 
-  after(:each) do
-    unless @latest_output.nil? || should_mock_response
+  let(:min_delete_params) do
+    {
+      '--compartment-id' => compartment_id,
+      '--bmcs-config-file' => config_file_path,
+      '--bmcs-profile' => profile,
+      '--yes' => true
+    }
+  end
+
+  it 'can create an Oracle Linux instance with min params' do
+    params = min_params
+    params['--image-id'] = oracle_linux_image_id
+    puts params.inspect
+    shell = run_server_create(params, 'test_oracle_linux_min_params')
+    validate_output(shell, params)
+
+    unless @latest_output.nil?
+      # delete the instance using default chef node name
       output = @latest_output
       @latest_output = nil
-
       match = output.match("Instance ID:\s(.*)")
       if match && match.length > 1
         instance_id = match[1]
+        chef_node_name = output.match("Bootstrapping with node name '(.+)'")[1]
         puts "Clean Up: Terminating instance #{instance_id}."
-        client = OracleBMC::Core::ComputeClient.new(config: OracleBMC::ConfigFileLoader.load_config(config_file_location: config_file_path, profile_name: profile))
-        client.terminate_instance(instance_id)
+        params = min_delete_params
+        params['--instance-id'] = instance_id
+        params['--purge'] = true
+        shell = run_server_delete(params, 'test_oracle_linux_delete_with_purge')
+        validate_delete_output(shell, chef_node_name)
       end
     end
   end
 
-  it 'can create an Oracle Linux instance with min params' do
-    params = min_params
-    params['--image-id'] = oracle_linux_image_id
-    shell = run_server_create(params, 'test_oracle_linux_min_params')
-    validate_output(shell, params)
-  end
-
   it 'can create an Oracle Linux instance with all params' do
     params = min_params.merge(extra_params)
     params['--image-id'] = oracle_linux_image_id
     shell = run_server_create(params, 'test_oracle_linux_all_params')
     validate_output(shell, params)
     expect(shell.stdout).to include(params['--display-name'])
+    unless @latest_output.nil?
```

Comment:
A bit of a nitpick, but this block is repeated 3 times. Seems worth moving into a method.